packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -190,7 +187,14 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 			}
 
 			return attributes;
-		}, [ blockBindings, name, clientId, blockContext, registry, sources ] );
+		}, [
+			blockBindings,
+			name,
+			clientId,
+			updatedContext,
```

Comment:
I think this should be passing `context`, and the selector should return the updated context together with attributes:

```js
const { boundAttributes, updatedContext } = useSelect( () => {
```

Reply:
It seems moving `updatedContext` inside the `useSelect` triggers the same issue reported [here](https://github.com/WordPress/gutenberg/pull/64072#discussion_r1764867975) where it throws the warning:

```
The 'useSelect' hook returns different values when called with the same state and parameters. This can lead to unnecessary rerenders.
```

I can take a look at the solutions suggested there, but I was wondering first: Would it be safe to directly modify the context prop? Instead of doing `updatedContext[ key ] = blockContext[ key ];` do directly `context[ key ] = blockContext[ key ];`.

Reply:
It's probably bad idea to mutate anything that React controls - props, context, etc.

Reply:
I've started exploring moving the updatedContext to a `useMemo` call in [this pull request](https://github.com/WordPress/gutenberg/pull/65604).